book/machinelearning.qmd

```diff
@@ -4,114 +4,215 @@ abstract: TODO (150-200 WORDS)
-
-:::: {.callout-note icon=false}
+## Evaluating and benchmarking {#sec-ml-eval}
+
+To understand if a model is 'good', its predictions are evaluated with a *loss function*.
+Loss functions assign a score to the discrepancy between predictions and true values, $L: \calY \times \calY \rightarrow \ExtReals$.
+Given real-world data, $(\XX^*, \yy^*)$, and a trained model $\hatf$, then the loss is given by $L(\hatf{\XX^*}, \yy^*) = L(\hatyy, \yy^*)$.
+For a model to be useful, it should perform well in general, and not just for the data used for training and development, which is known as a model's *generalization error*.
+
+A model should not be deployed, i.e., manually or automatically used to make predictions, its generalization error must be estimated to understand if the model is 'good'.
+If a model were to be trained and evaluated on the same data, the resulting loss, known as the *training error*, would be an overoptimistic estimate of the true generalization error (@Hastie2013) as the model is making predictions for data it has already 'seen' and the loss is not evaluating its ability to generalize to new, unseen data.
+Estimation of the generalization error requires *data splitting*, which is the process of splitting available data, $\calD$, into *training data*, $\dtrain \subset \calD$, and *testing data*, $\dtest = \calD \setminus \dtrain$.
+<!-- FIXME: ABOVE COULD BE BETTER WORDED -->
+
+The simplest method to estimate the generalization error is to use *holdout resampling*, which is the process of partitioning the data into one training dataset and one testing dataset, with the model trained on the former and predictions made for the latter.
+Using 2/3 of the data for training and 1/3 for testing is a common splitting ratio (@Kohavi1995).
+When splitting, it is essential the data is partitioned randomly to ensure any information encoded in data ordering is removed.
+Ordering is often important in the real-world, for example in healthcare data when patients are recorded in order of enrolment to a study.
+Whilst ordering can provide useful information, it does not generalize to new, unseen data.
+For example, the number of days a patient has been in hospital is more useful than the patient's index in the dataset, as the former could be calculated for a new patient whereas the latter is meaningless.
+Another example is in the `rats` dataset that was briefly described in @sec-surv.
+The `rats` data explores how a novel drug effects tumor incidence.
+However, the data is ordered by rat litters with every three rats being in the same litter.
+Hence if rats were to be bred or raised differently over time,  even if the `litter` column were removed, this information would still be encoded in the order of the dataset and could impact upon any findings.
+Randomly splitting breaks any possible association between order and outcome.
```

Comment:
There are situations where random splitting is not good: Time-series, hierarchical data (where rats litter could be an example for), spatially correlated data, etc.. I think i would write that in iid case random sampling is good, in other contexts other strategies might be helpful -> cite our overview paper http://arxiv.org/abs/2310.15108.